lib/logger/lib/logger/backends/console.ex

```diff
@@ -72,11 +102,27 @@ defmodule Logger.Backends.Console do
       enabled: Keyword.get(colors, :enabled, IO.ANSI.enabled?)}
   end
 
-  defp log_event(level, msg, ts, md, %{colors: colors, device: device} = state) do
+  defp log_event(level, msg, ts, md, state) do
+    %{colors: colors, device: device} = state
+    output =
+      format_event(level, msg, ts, md, state)
+      |> color_event(level, colors)
+    %{state | ref: io_request(device, output)}
+  end
+
+  defp io_request(device, output) do
+    ref = make_ref
+    send(device, {:io_request, self(), ref, {:put_chars, :unicode, output}})
+    ref
+  end
+
+  defp buffer_event(level, msg, ts, md, state) do
+    %{colors: colors, buffer: buffer, batch_size: batch_size} = state
     output =
       format_event(level, msg, ts, md, state)
       |> color_event(level, colors)
-    send(device, {:io_request, self(), self(), {:put_chars, :unicode, output}})
+    buffer = [buffer | output]
+    %{state | buffer: buffer, batch_size: batch_size + 1}
```

Comment:
Isn't it weird we are buffering events even when we know we will discard them later on?